### ISSUE TITLE
Problem with is_local in system.clusters table

### ISSUE THREAD

**Issue by bgranvea** [Altinity/clickhouse-operator#138]
I have a cluster of 2 shards / 2 replicas deployed on real machines and another one deployed on Kubernetes with Clickhouse-operator. I noticed a strange difference in the value of is_local in system.clusters table. It is a problem as our application relies on this to determine the shard it is connected to.

Kubernetes:
```
# kubectl -n test exec -it chi-dev-vscluster-0-0-0 -- clickhouse-client -q "select shard_num,replica_num,host_name,host_address,is_local from system.clusters where cluster='vscluster'"
1       1       chi-dev-vscluster-0-0   10.244.2.84     1
1       2       chi-dev-vscluster-0-1   10.244.0.64     1
2       1       chi-dev-vscluster-1-0   10.244.2.85     0
2       2       chi-dev-vscluster-1-1   10.244.2.86     0
# kubectl -n test exec -it chi-dev-vscluster-0-1-0 -- clickhouse-client -q "select shard_num,replica_num,host_name,host_address,is_local from system.clusters where cluster='vscluster'"
1       1       chi-dev-vscluster-0-0   10.244.2.84     1
1       2       chi-dev-vscluster-0-1   10.244.0.64     1
2       1       chi-dev-vscluster-1-0   10.244.2.85     0
2       2       chi-dev-vscluster-1-1   10.244.2.86     0
# kubectl -n test exec -it chi-dev-vscluster-1-0-0 -- clickhouse-client -q "select shard_num,replica_num,host_name,host_address,is_local from system.clusters where cluster='vscluster'"
1       1       chi-dev-vscluster-0-0   10.244.2.84     0
1       2       chi-dev-vscluster-0-1   10.244.0.64     0
2       1       chi-dev-vscluster-1-0   10.244.2.85     0 <= should be 1!
2       2       chi-dev-vscluster-1-1   10.244.2.86     0 <= should be 1!
# kubectl -n test exec -it chi-dev-vscluster-1-1-0 -- clickhouse-client -q "select shard_num,replica_num,host_name,host_address,is_local from system.clusters where cluster='vscluster'"
1       1       chi-dev-vscluster-0-0   10.244.2.84     0
1       2       chi-dev-vscluster-0-1   10.244.0.64     0
2       1       chi-dev-vscluster-1-0   10.244.2.85     0 <= should be 1!
2       2       chi-dev-vscluster-1-1   10.244.2.86     0 <= should be 1!
```

Real machines:
```
# clickhouse-client -q "select shard_num,replica_num,host_name,host_address,is_local from system.clusters where cluster='vscluster'"
1       1       dca-test2.iv.local      10.148.20.42    1
1       2       dca-test4.iv.local      10.148.20.44    1
2       1       dca-test3.iv.local      10.148.20.43    0
2       2       dca-test5.iv.local      10.148.20.49    0
# clickhouse-client -q "select shard_num,replica_num,host_name,host_address,is_local from system.clusters where cluster='vscluster'"
1       1       dca-test2.iv.local      10.148.20.42    0
1       2       dca-test4.iv.local      10.148.20.44    0
2       1       dca-test3.iv.local      10.148.20.43    1
2       2       dca-test5.iv.local      10.148.20.49    1
# clickhouse-client -q "select shard_num,replica_num,host_name,host_address,is_local from system.clusters where cluster='vscluster'"
1       1       dca-test2.iv.local      10.148.20.42    1
1       2       dca-test4.iv.local      10.148.20.44    1
2       1       dca-test3.iv.local      10.148.20.43    0
2       2       dca-test5.iv.local      10.148.20.49    0
# clickhouse-client -q "select shard_num,replica_num,host_name,host_address,is_local from system.clusters where cluster='vscluster'"
1       1       dca-test2.iv.local      10.148.20.42    0
1       2       dca-test4.iv.local      10.148.20.44    0
2       1       dca-test3.iv.local      10.148.20.43    1
2       2       dca-test5.iv.local      10.148.20.49    1

```
This is the YAML I used to deploy:
```
apiVersion: "clickhouse.altinity.com/v1"
kind: "ClickHouseInstallation"
metadata:
  name: "dev"
spec:
  defaults:
    templates:
      volumeClaimTemplate: volumeclaim-template
  configuration:
    zookeeper:
      nodes:
      - host: zookeeper.test
        port: 2181
    clusters:
      - name: "vscluster"
        layout:
          shardsCount: 2
          replicasCount: 2
  templates:
    volumeClaimTemplates:
      - name: volumeclaim-template
#        reclaimPolicy: Retain
        spec:
          accessModes:
            - ReadWriteOnce
          resources:
            requests:
              storage: 10Gi
```

Do you have an idea why is_local=0 in some of the containers?

**Comment by alex-zaitsev** [comment]
Thank you, we will look into that. It seems related to DNS resolution that is not instant. In my environment the problem disappeared once I manually executed 'SYSTEM DROP DNS CACHE' on affected hosts. New ClickHouse versions (19.12 and above) should have better DNS cache management, so the issue is probably fixed by then automatically.

**Comment by bgranvea** [comment]
it works with your workaround, thanks!

**Comment by alex-zaitsev** [comment]
Fixed in 0.4.4. Now pod starts and restarts correctly initialising is_local.
